include/lbann/layers/transform/scatter.hpp

```diff
@@ -128,31 +162,81 @@ El::Device scatter_layer<TensorDataType,Layout,Device>::get_device_allocation()
 template <typename TensorDataType, data_layout Layout, El::Device Device>
 void scatter_layer<TensorDataType,Layout,Device>::setup_dims(DataReaderMetaData& dr_metadata) {
   data_type_layer<TensorDataType>::setup_dims(dr_metadata);
+  
 
-  // Tensor dimensions
   const auto& input0_dims = this->get_input_dims(0);
   const auto& input1_dims = this->get_input_dims(1);
+  const auto& output_dims = this->get_output_dims();
+
+  auto dims_to_str = [] (const std::vector<int>& dims) -> std::string {
+    std::ostringstream ss;
+    for (size_t i=0; i<dims.size(); ++i) {
+      ss << (i>0 ? "x" : "") << dims[i];
+    }
+    return ss.str();
+  };
```

Comment:
Don't we have a function for this? If no, we should have one at the library utils level...

Reply:
Not that I know of, but this pattern does get used quite a bit. Ex:
https://github.com/LLNL/lbann/blob/4acbbe5fcb0b48af35a1afa48e628d789ecaf0f1/include/lbann/layers/transform/concatenate.hpp#L226

https://github.com/LLNL/lbann/blob/4acbbe5fcb0b48af35a1afa48e628d789ecaf0f1/include/lbann/layers/transform/sum.hpp#L116

Reply:
That's fantastic. Not your problem, but we should probably functionify that...